src/main/java/org/rumbledb/expressions/control/TypeswitchCase.java

```diff
@@ -0,0 +1,59 @@
+package org.rumbledb.expressions.control;
+
+
+import org.rumbledb.expressions.Expression;
+import org.rumbledb.expressions.flowr.FlworVarSequenceType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is a helper class that organizes the children expressions of a typeswitch expression.
+ * From a tree perspective, all expressions in there are considered
+ * to be direct children of the TypeswitchExpression.
+ */
+public class TypeswitchCase {
+
+    private String variableName;
+    private List<FlworVarSequenceType> union = new ArrayList<>();
+    private final Expression returnExpression;
+
+    public TypeswitchCase(
+            String variableName,
+            List<FlworVarSequenceType> union,
+            Expression returnExpression
+    ) {
+        this.variableName = variableName;
+        this.union.addAll(union);
+        this.returnExpression = returnExpression;
+    }
+
+    // for default
```

Comment:
Very tiny remark. As code evolves this comment may become outdated and even misleading. I believe with modern IDEs developers could easily lookup where this constructor is used.

Reply:
This is now removed.